report_thesis/src/sections/proposed_approach/testing_validation.tex

```diff
@@ -47,76 +59,142 @@ \subsection{Validation and Testing Procedures for Model Evaluation}\label{subsec
+\For{each fold $i$ from 0 to $k-2$}
+    \State $\mathbf{T}_\text{train}[i] \gets \mathbf{D}_\text{train} \setminus \text{fold}_i$
+    \State $\mathbf{T}_\text{val}[i] \gets \text{fold}_i$
+    \State Append $(\mathbf{T}_\text{train}[i], \mathbf{T}_\text{val}[i])$ to $\mathbf{T}_\text{cv}$
 \EndFor
 \State \label{line:remove_fold_column} Remove fold column from all datasets
-\State \Return \( \mathbf{F}_\text{cv}, \mathbf{D}_\text{train}, \mathbf{D}_\text{test} \)
+\State \Return $\mathbf{T}_\text{cv}, \mathbf{D}_\text{train}, \mathbf{D}_\text{test}$
 \end{algorithmic}
 \end{algorithm}
 
 The procedure outlined in Algorithm~\ref{alg:custom_kfold_cv} begins by setting a random seed for reproducibility if one is provided (Line~\ref{line:seed}).
 This ensures that the results are consistent across different runs of the algorithm.
-Next, the dataset is processed to remove any duplicate entries based on the group column and then sorted by the target column (Line~\ref{line:remove_duplicates}).
+Next, the dataset is processed to remove any duplicate entries based on the group column $g$ and then sorted by the target column $t$ (Line~\ref{line:remove_duplicates}).
 This step ensures that each group is uniquely identified and ordered appropriately.
 The dataset we illustrate in Table~\ref{tab:final_dataset_example} would require a group column $g$ of "\texttt{Target}" to group the data by target.
 The target column $t$ refers to the column with the target variable, which would be the oxide for which we are predicting the concentration, for example, \ce{SiO_2}.
-By sorting the dataset by the target column, we ensure that the data is ordered by the target concentration values in ascending order.
+By sorting the dataset by the target column $t$, we ensure that the data is ordered by the target concentration values in ascending order.
 
 Fold numbers are then assigned sequentially using a modulo operation to ensure a random-like distribution of the unique targets across the folds (Line~\ref{line:assign_folds}).
 This means that, while the assignment process follows a sequence, the resulting distribution of targets is effectively randomized.
-If handling of extreme values is enabled, the algorithm identifies the top and bottom percentiles of the target values (Line~\ref{line:identify_extremes}) and reassigns these extreme values to the training folds (0 to \( k-2 \)) to ensure they are well-represented during model training (Line~\ref{line:reassign_extremes}).
+Fold numbers start in 0 and go up to $k-1$, as implied by the modulo operation.
+If a percentile $p$ is provided, extreme value handling is enabled, and the algorithm identifies the top and bottom percentiles of the target values (Line~\ref{line:identify_extremes}).
+These extreme values are then reassigned to the training folds (0 to $k-2$), ensuring they are evenly distributed across these folds (Line~\ref{line:reassign_extremes}).
```

Comment:
I prefer to avoid fancy words for simplicity. What about this?

```suggestion
These extreme values are then reassigned to the training folds (0 to $k-2$), ensuring they are as evenly distributed as possible across these folds (Line~\ref{line:reassign_extremes}).
```